content/tutorials/mqtt-snake.textile

```diff
@@ -0,0 +1,539 @@
+    go get -u github.com/nsf/termbox-go
+  ```
+
+  You'll now have access to these libraries when using Go.
+
+blang[python].
+  You'll also need the "paho-mqtt":https://pypi.python.org/pypi/paho-mqtt/1.3.1 library for the controller to use MQTT, so type the following into your command line to get it:
+
+  ```[sh]
+    pip install paho-mqtt
+  ```
+
+h2. Step 3 - Add your libraries
+
+blang[javascript].
+  With your NPM modules ready, it's time to create a file to contain your controller code. Create a file called @controller.js@ inside your @mqtt-snake@ folder. Inside this file, instance the MQTT.js and keypress libraries:
+
+  ```[javascript]
+    const mqtt = require('mqtt');
+    const keypress = require('keypress');
+  ```
+
+  "See this step in Github":https://github.com/ably/tutorials/commit/mqtt-snake-js-step3
+
+blang[go].
+  With your libraries installed, it's time to create a file to contain your controller code. Create a file called @controller.go@ inside your @mqtt-snake@ folder. Inside this file, reference the libraries we'll need:
+
+  ```[go]
+    package main
+
+    import (
+      'fmt'
+      MQTT 'github.com/eclipse/paho.mqtt.golang'
+      termbox 'github.com/nsf/termbox-go"
+    )
+  ```
+
+  "See this step in Github":https://github.com/ably/tutorials/commit/mqtt-snake-go-step3
+
+blang[python].
+  With your libraries ready, it's time to create a file to contain your controller code. Create a file called @controller.py@ inside your @mqtt-snake@ folder. Inside this file, instance the paho-mqtt and curses libraries:
+
+  ```[python]
+    import paho.mqtt.client as mqtt
+    import curses
+  ```
+
+  "See this step in Github":https://github.com/ably/tutorials/commit/mqtt-snake-python-step3
+
+h2. Step 4 - Connect to Ably through MQTT
+
+With the libraries now available, it's time to set up MQTT. When using MQTT with Ably, there are a few requirements with regards to your setup. You'll need to connect to 'mqtt.ably.io' on port 8883, which requires the use of SSL/TLS in your connection. If you have a device which cannot support SSL, you'll need to connect via port 1883 instead, but this will come with a number of "restrictions":https://support.ably.io/solution/articles/3000074906-using-the-mqtt-protocol-adapter. 
+
+You will need to set the "keep alive" time value to between 15 and 60 seconds, where 60 seconds will maximize the battery life, and 15 seconds will maximize responsiveness to network issues. For this tutorial you'll be setting it to 15.
+
+For authentication, you'll need to either use an API Key, or a token. For simplicity of the tutorial, you'll be using the API key you obtained in step 1. For this, you'll need to provide the first half of the API Key (before the colon) as the username, and the second half of the API Key (after the colon) as the password. For example, if your API key is @A12B3C.4D:5E6F7G8H@, your username will be @A12B3C.4D@ and your password will be @5E6F7G8H@.
```

Comment:
I agree for this tutorial an API key is good, but we need to provide a link for users to ready why we rarely recommend a user uses an API key on an untrusted device.  Ideally we'd have an MQTT article (help document) specifically on authentication which explains this and the options available to them, including a simple example where an MQTT device obtains a token from an endpoint before connecting, and handles disconnects and gets a new token for example.